dissect/target/target.py

```diff
@@ -285,10 +282,14 @@ def _find(find_path: Path):
                         # Attempt to load the target using this loader
                         target = cls._load(sub_entry, ldr)
                         loaded = True
+                        at_least_one_loaded = True
                         yield target
-                    except Exception as e:
-                        getlogger(sub_entry).error("Failed to load target with loader %s", ldr, exc_info=e)
-                        continue
+
+                        # If DirLoader, ignore further entries. Otherwise search children/nested
+                        if isinstance(ldr, DirLoader):
```

Comment:
Instinctively this break feels like it's one indent too far. I know that `DirLoader` currently has no custom `find_all()` implemented, but if it would, would this not break after the first entry?

Reply:
Yes, this is intended. It is not desirable for the DirLoader to generate subentries as the dir as a whole is loaded, this is almost by definition I would say. I think that generating subentries for the Dirloader would make things quite complex because then the DirLoader either encompasses the whole dir or it loads all subdirs and this may interfere with other loaders and files to be ignored. I think it would be better to simplify at this point. But maybe we do not understand each other correctly or I am missing something. Feel free to correct me.

Reply:
It's half that it feels like the code would be more correct with it being one indent level less (but still have the same effect, in the current state of the code base), and half that it may have a side-effect in a future loader scenario.

Currently, `DirLoader`'s `find_all()` implementation only yields itself. This is correct. However, say a hypothetical situation and format exists that would look like this:

```
/root/metadata.bin
/root/data/host1/<dir structure>
/root/data/host2/<dir structure>
/root/data/host3/<dir structure>
```

In this hypothetical situation, `metadata.bin` would contain important information that is necessary to load data from `host1`, `host2` and `host3`. For example, a decryption key. A possible loader for this could subclass from `DirLoader` and could implement the `detect()` method to look for the existence of `metadata.bin`, so `/root` is a valid target for this loader. However, it actually contains multiple "targets", so its `find_all()` implementation would iterate from the `/root/data` directory.

With the break being where it is now, only `/root/data/host1` would get returned.

Of course there are many other different questions and problems with this scenario (why subclass from `DirLoader`? why not create a `detect()` that looks for a `metadata.bin` in its parent directories?) but it does highlight why I believe it should be one indent less.

Then again, of course, I could also be overlooking something why one indent less would break :smile:

Reply:
Ok, convinced, thanks for your explanation, this seems to make sense.